planner/core/physical_plans.go

```diff
@@ -131,6 +131,27 @@ type PartitionInfo struct {
 	ColumnNames    types.NameSlice
 }
 
+const emptyPartitionInfoSize = int64(unsafe.Sizeof(PartitionInfo{}))
+
+// MemoryUsage return the memory usage of PartitionInfo
+func (pi *PartitionInfo) MemoryUsage() (sum int64) {
+	if pi == nil {
+		return
+	}
+
+	sum = emptyPartitionInfoSize
+	for _, cond := range pi.PruningConds {
+		sum += cond.MemoryUsage()
+	}
+	for _, cis := range pi.PartitionNames {
+		sum += cis.MemoryUsage()
+	}
+	for _, col := range pi.Columns {
+		sum += col.MemoryUsage()
+	}
+	return
```

Comment:
Should we accumulate memory usage of `ColumnNames` here?

Reply:
> Should we accumulate memory usage of `ColumnNames` here?

yes, you are right